Source/UI/NewUI/PaymentForm/Factory/PaymentFormFactory.swift

```diff
@@ -15,13 +15,15 @@ public struct PaymentFormFactory {
     let cardValidator = CardValidator(environment: configuration.environment.checkoutEnvironment)
     let checkoutAPIService = CheckoutAPIService(publicKey: configuration.serviceAPIKey,
                                                 environment: configuration.environment)
-    let viewModel = DefaultPaymentViewModel(checkoutAPIService: checkoutAPIService,
+    var viewModel = DefaultPaymentViewModel(checkoutAPIService: checkoutAPIService,
                                             cardValidator: cardValidator,
                                             logger: logger,
                                             billingFormData: configuration.billingFormData,
                                             paymentFormStyle: style.paymentFormStyle,
                                             billingFormStyle: style.billingFormStyle,
                                             supportedSchemes: configuration.supportedSchemes)
+    viewModel.preventDuplicateCardholderInput()
```

Comment:
Question : Is `preventDuplicateCardholderInput` checking if Billing Form has another card holder text field and removes it ? if so do we need to really restrict this from SDK.

Reply:
> Question : Is preventDuplicateCardholderInput checking if Billing Form has another card holder text field and removes it

The changes inside `PaymentViewModel.swift` are implementing this logic. Simply put, if payment form has a cardholder name field, it will remove a fullName field from billing form cells. This is so we do not ask multiple times for the same property, either having a property we discard or allowing the user to change the same thing from 2 places. 

@patrick-hoban-cko has responded to this [on Jira task](https://checkout.atlassian.net/browse/PIMOB-1379?focusedCommentId=182585), with outcome:
_Agree that a merchant choosing the cardholder name to show on the payment form should supersede the billing address._


***


> if so do we need to really restrict this from SDK.

I'm not sure what you mean? The method is internal and consumer cannot call it. The call site is also part of the factory screen creation and should not expose anything to consumer that may allow them to call on demand, undo logic or in any way misuse it 🤔

Reply:
Looking at the [API spec](https://api-reference.checkout.com/#operation/requestAToken) and also old UI. Should we not just add the card holder name to payment form always to keep it simple and not have additional logic to remove it from billing form screen. Please correct me if I have missed anything here.

Reply:
I don't think you missed anything and you might actually have a better knowledge of Frames altogether so you might know why it has been implemented to display full name within billing to begin with? I am afraid I don't have an answer from my part, didn't see discoveries, plannings or tickets on that 🤷‍♂️. 

From my side I am only trying to avoid bad UX, but if you're thinking there's another path to it I am happy to discuss on how we can achieve.